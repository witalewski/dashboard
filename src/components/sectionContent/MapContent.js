import React, { Component } from 'react';
import styled from '@emotion/styled';
import { chartColors } from '../../global/styleConstants';
import { iso2ToIso3 } from '../../utils/countryCodeConverter';
import * as Datamap from 'datamaps';

const MapContentStyled = styled.div`
  min-width: 800px;
  min-height: 400px;

  display: flex;
  flex-direction: column;
  align-items: stretch;

  .map-container {
    position: relative;
    flex: 1 1 100%;
  }

  .tooltip {
    margin: 0px;
    padding: 10px;
    background-color: rgb(255, 255, 255);
    border: 1px solid rgb(204, 204, 204);
    font-size: 14px;
    white-space: nowrap;
  }
`;

export class MapContent extends Component {
  componentDidMount() {
    // Should only be run in the browser, not in test environment
    // Datamap causes error in jest: [TypeError: Datamap is not a constructor]
    if (typeof jest === 'undefined') {
      new Datamap({
        element: document.getElementsByClassName('map-container')[0],
        fills: {
          MARKED: chartColors[0],
          defaultFill: '#dddddd',
        },
        data: this.props.data.reduce(
          (acc, { label, tooltip }) => ({
            ...acc,
            [iso2ToIso3(label)]: { fillKey: 'MARKED', tooltip },
          }),
          {}
        ),
        geographyConfig: {
          highlightFillColor: chartColors[2],
          popupTemplate: function(geo, data) {
            return [
              '<div class="tooltip"><p><strong>',
              geo.properties.name,
              '</strong></p><p>',
              data.tooltip.replace(/\n/g, '<br/>'),
              '</p></div>',
            ].join('');
          },
        },
      });
    }
  }
  render() {
    return (
      <MapContentStyled>
        <div className="map-container" />
      </MapContentStyled>
    );
  }
}
