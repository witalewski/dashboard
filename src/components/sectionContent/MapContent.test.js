import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockChartData from '../../fixtures/mockChartData.json';
import { MapContent } from './MapContent';

describe('MapContent', () => {
  it('matches snapshot with mock data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <MapContent
        title="Geography"
        className="revenue-section-wrapper"
        data={mockChartData.mapChart}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
