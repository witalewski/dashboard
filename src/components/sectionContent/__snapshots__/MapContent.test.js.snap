// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContent matches snapshot with mock data 1`] = `
<ForwardRef(render)>
  <div
    className="map-container"
  />
</ForwardRef(render)>
`;
